scipy/optimize/lbfgsb.py

```diff
@@ -218,8 +218,9 @@ def _minimize_lbfgsb(fun, x0, args=(), jac=None, bounds=None,
 
     Options
     -------
-    disp : bool
-       Set to True to print convergence messages.
+    disp : int
+        If zero, then no output.  If a positive number, then this over-rides
```

Comment:
If `disp is None` (the default), then the supplied version of `iprint` is used. If `disp is not None`, then it overrides the supplied version of `iprint` with the behaviour you outlined.